idaes/models/unit_models/stream_scaler.py

```diff
@@ -0,0 +1,239 @@
+
+    def initialize_build(
+        blk, outlvl=idaeslog.NOTSET, optarg=None, solver=None, hold_state=False
+    ):
+        """
+        Initialization routine for StreamScaler.
+
+        Keyword Arguments:
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None, use
+                     default solver options)
+            solver : str indicating which solver to use during
+                     initialization (default = None, use default solver)
+            hold_state : flag indicating whether the initialization routine
+                     should unfix any state variables fixed during
+                     initialization, **default** - False. **Valid values:**
+                     **True** - states variables are not unfixed, and a dict of
+                     returned containing flags for which states were fixed
+                     during initialization, **False** - state variables are
+                     unfixed after initialization by calling the release_state
+                     method.
+
+        Returns:
+            If hold_states is True, returns a dict containing flags for which
+            states were fixed during initialization.
+        """
+        # init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        # solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        # Create solver
+
+        # Initialize inlet state blocks
+        flags = blk.inlet_block.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            hold_state=True,
+        )
+
+        if hold_state is True:
+            return flags
+        else:
+            blk.release_state(flags, outlvl=outlvl)
+
+    def release_state(blk, flags, outlvl=idaeslog.NOTSET):
+        """
+        Method to release state variables fixed during initialization.
+
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state = True.
+            outlvl : sets output level of logging
+
+        Returns:
+            None
+        """
+        blk.inlet_block.release_state(flags, outlvl=outlvl)
+
+    def _get_stream_table_contents(self, time_point=0):
+        io_dict = {
+            "Inlet": self.inlet,
+            # "Outlet": self.outlet,
+        }
+        return create_stream_table_dataframe(io_dict, time_point=time_point)
+
+    def calculate_scaling_factors(self):
+        # Scaling factors for the property block are calculated automatically
+        super().calculate_scaling_factors()
+
+        # Need to pass on scaling factors from the property block to the outlet
+        # VarLikeExpressions so arcs get scaled right
+        scale = 1 / self.multiplier.value
```

Comment:
You could also use a Python idiom for simple if/else assignments (1 line instead of 4):
`default = 1 if self.multiplier.value == 0 else 1 / self.multiplier.value`